app/controllers/ATSPAYEDataController.scala

```diff
@@ -42,17 +42,25 @@ class ATSPAYEDataController @Inject()(npsService: NpsService, payeAuthAction: Pa
 
   def getATSDataMultipleYears(nino: String, yearFrom: Int, yearTo: Int): Action[AnyContent] = payeAuthAction.async {
     implicit request =>
-      def dataList: Seq[Future[Option[JsValue]]] = (yearFrom to yearTo).toList map { year =>
+      def dataList: Seq[Future[Either[Int, JsValue]]] = (yearFrom to yearTo).toList map { year =>
```

Comment:
Could just keep the Int as a HttpResponse here, saves you unpacking it to an Int

Reply:
This makes it harder when it comes to line 60. If we do .contains(Left(HttpResponse)) it never matches for some reason when using HttpResponse, it works when using an Int however.

Reply:
Ignore my comments, there are just informative, follow Ste comments. There is a fundamental issue with returning an Int: You have no context.
2 years later in an obscure service you get an Int from a method. What is that Int about? You don't know.

Compare to what you get from http_verbs (http.GET) in case of an error. You receive something named UpstreamErrorResponse. Notice that you immediately see it's an error and its a response to an upstream service. See how much information the type gives you, scala makes this kind of patterns easy. Especially that with Either you can have a different type for your error and your success.

Now, you have 2 options to do something similar. Build an error object from scratch. The coolTool does that, not the best implementation to be honest but this is the idea: https://github.com/hmrc/live-incident-support-tool/blob/master/app/models/CoolToolErrors.scala
Or you use an exiting one and that UpstreamErrorResponse looks a good candidate.

I am not a big fan of using HttpResponse here because of the context it carries but technically it is correct.